niworkflows/interfaces/cifti.py

```diff
@@ -355,7 +356,7 @@ def _create_cifti_image(
             hemi = structure.split("_")[-1]
             # currently only supports L/R cortex
             surf = nb.load(bold_surfs[hemi == "RIGHT"])
-            surf_verts = len(surf.darrays[0].data)
+            surf_verts = len(surf.agg_data()[0])
```

Comment:
Do not assume order. Try something like:

```suggestion
            surf_verts = len(img.agg_data('pointset'))
```

Reply:
order shouldn't matter here, it's just polling the number of vertices of a given data array

Reply:
You're using `[0]`. A surface file is a pointset and a triangles. These don't have the same shape.

Reply:
These are functional (`func.gii`) GIFTIs, which I believe don't have any triangles included?

(Looking back, `surf` probably wasn't the best variable name).

Reply:
If that is the case (faces are not included), what is the reason for doing so? If we are dropping the triangles, I believe we should re-add them. There's no value in dropping them.

Reply:
Not sure this change is worth making, then. It makes a copy of all the data just for the purpose of taking the length of the first dimension.

Reply:
Oscar, this is a time series. They don't have coordinates and faces.

Reply:
GIFTI does not impose what you can't embed on the file. NIfTI is great in the sense that, you can access the data and reconstruct the sampling structure. If vertices and faces are not stored along with the time-series, it is definitely not self-contained and also prone to failures. Whether the visualization software does not check whether the surface is packaged within the CIFTI, that's a different issue. But I believe we should try to generate self-contained derivatives whenever that's possible.

Reply:
Do any tools accept this?

Reply:
FWIW, CIFTIs are not intended to store surface information - generally surface geometry information is supplemental

Reply:
> Do any tools accept this?

Okay, after reflecting a bit on this - there's the problem that one of these timeseries may map on to several surfaces (midthickness, inflated, sphere...). I guess, in the extreme, you'd want to keep the information of triangles. But then you'll always need the pointset so it's kind of pointless (did you see what I did there?). I guess there's no option but just not storing surface information.



> CIFTIs are not intended to store surface information

Yup, but we are talking about GIFTI here, right?